### PR TITLE
add business logic in create-news

### PR DESCRIPTION
**Pull request by Gusvarov**

### CODE REVIEW

File: src/app/component/general/eco-news/create-news/create-news.component.css [hunk 1]
```diff
@@ -27,6 +37,14 @@
...
+    border: 1px solid red !important;
+}
+
+input.ng-invalid.ng-touched + span, textarea.ng-invalid.ng-touched + p {
+    color: red;
+}
+
 .title h2 {
     font-size: 48px;
     font-family: 'Montserrat', sans-serif;
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.css]: Please rename 'info' class, 'cause it's not clear

File: src/app/component/general/eco-news/create-news/create-news.component.html [hunk 2]
```diff
@@ -3,9 +3,11 @@
...
-        <span>Ukrainian</span>
-        <span>English</span>
-        <span>Russian</span>
+        <span (click)="changeLanguage(language)" 
+               *ngFor="let language of languages" 
+               [ngClass]="{
+                   active: language.lang === activeLanguage
+                }">{{ language.name }}</span>
     </div>
     <div class="info">
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.html]: Rename 'info' class. It's not clear

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 3]
```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
...
 
-  constructor() { }
+  public createNewsForm = this.fb.group({
+    title: ['', [Validators.maxLength(170), Validators.required]],
+    source: [''],
+    content: ['', [Validators.minLength(20), Validators.required]],
+    tags: this.fb.array([])
+  });
+
+  public filters: Array<any> = [
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please specify type in generic

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 4]
```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
...
+    {name: 'Initiatives', isActive: false}, 
+    {name: 'Ads', isActive: false}
+  ];
+
+  public activeLanguage: string = 'en';
+  public link: string;
+  private preparedLink: string = preparedImageForCreateEcoNews;
+  private date: Date = new Date();
+
+  public languages: Array<any> = [
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: The same as line 25

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 5]
```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
...
   ngOnInit() {
+    this.link = this.preparedLink;
+    this.onChanges();
+  }
+
+  private navigateByUrl(): void {
+    this.router.navigateByUrl('/news');
+  }
+
+  public onChanges(): void {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please rename this method 'cause it's not clear

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 6]
```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
...
+    this.createNewsForm.get('source').valueChanges.subscribe(source => {
+      if(!source) {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: bad code style

Comment (nabermi2020) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: please fix it

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 7]
```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
...
+        this.link = this.preparedLink;
+      } else {
+        this.link = source;
+      }
+    });
   }
 
-}
+  public onSubmit(): void {
+    console.log(this.createNewsForm.value);
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please replace console.log with // TO DO

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 8]
```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
...
+      this.createNewsForm.value.tags.forEach((item, index) => {
+        if (item === filter.name) {
+          this.createNewsForm.value.tags.splice(index, 1);
+        }
+      })
+    }
+  }
+
+  public changeLanguage(language: any): void {
+    let formData = {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Use const instead of let

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 9]
```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
...
+      text: this.createNewsForm.value['content'],
+      title: this.createNewsForm.value['title']
+    }  
+
+    this.createEcoNewsService.setTranslationByLang(this.activeLanguage, formData);
+
+    this.activeLanguage = language.lang;
+    let translationData = this.createEcoNewsService.getTranslationByLang(this.activeLanguage);
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: the same as line 82

Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Extract the code from line 90 to 94 into a separate method

File: src/app/service/eco-news/create-eco-news.service.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      text: "",
+      title: ""
+    },
+    'en': {
+      text: "",
+      title: ""
+    }
+  }
+
+  public getTranslationByLang(lang: string): {text: string, title: string } {
```
Comment (nabermi2020) [inline src/app/service/eco-news/create-eco-news.service.ts]: Please create appropriate interface for { text: string, title: string }

File: src/app/component/general/eco-news/create-news/create-news.component.css (hunk 1 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.css]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.html (hunk 2 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.html]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 3 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 4 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 5 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 6 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 8 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 9 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/service/eco-news/create-eco-news.service.ts (hunk 10 above)
Comment (Gusvarov) [reply src/app/service/eco-news/create-eco-news.service.ts]: Created

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 7 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news-interface.ts [hunk 11]
```diff
@@ -0,0 +1,5 @@
+export interface CreateNewsInterface {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news-interface.ts]: Please rename it without using the verb in the interface name

File: src/app/component/general/eco-news/create-news/create-news.component.css [hunk 12]
```diff
@@ -157,6 +177,31 @@ form label {
...
+.submit-buttons button:last-child:disabled {
+    background-color: #839C94;
+    border: 1px solid #839C94;
+}
+
+button {
+    outline: none;
+}
+
+.green {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.css]: Please rename it, 'cause it's not clear

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 13]
```diff
@@ -1,18 +1,111 @@
...
 
-  constructor() { }
+  public createNewsForm = this.fb.group({
+    title: ['', [Validators.maxLength(170)]],
+    source: [''],
+    content: ['', [Validators.minLength(20)]],
+    tags: this.fb.array([])
+  });
+
+  public filters: Array<object> = [
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please specify the type in generic with more details. For example 'Array<{name: string, isActive: boolean}>'.

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 14]
```diff
@@ -1,18 +1,111 @@
...
+    {name: 'Ads', isActive: false}
+  ];
+
+  public activeLanguage: string = 'en';
+  public link: string;
+  private preparedLink: string = preparedImageForCreateEcoNews;
+  private date: Date = new Date();
+  public formValid: boolean;
+
+  public languages: Array<object> = [
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: the same as line 25

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 15]
```diff
@@ -1,18 +1,111 @@
...
+      } else {
+        this.link = source;
+      }
+    });
   }
 
-}
+  public onSubmit(): void {
+    this.createEcoNewsService.sendFormData(this.createNewsForm, this.activeLanguage).subscribe(
+      (data: any) => {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please, specify the argument type

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 16]
```diff
@@ -1,18 +1,111 @@
...
+        console.log(data);
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please, delete 'console.log'

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 17]
```diff
@@ -1,18 +1,111 @@
...
+      }
+    );
+  }
+
+  public addFilters(filter: any): void {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please, specify the argument type

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 18]
```diff
@@ -1,18 +1,111 @@
...
+  public onSubmit(): void {
+    this.createEcoNewsService.sendFormData(this.createNewsForm, this.activeLanguage).subscribe(
+      (data: any) => {
+        console.log(data);
+      }
+    );
+  }
+
+  public addFilters(filter: any): void { 
+    if(!filter.isActive) {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: bad formatting

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 19]
```diff
@@ -1,18 +1,111 @@
...
+
+  public setFormValue(): void {
+    const translationData = this.createEcoNewsService.getTranslationByLang(this.activeLanguage);
+    this.createNewsForm.patchValue({
+      title: translationData.title,
+      content: translationData.text
+    });
+  }
+
+  public changeLanguage(language: any): void {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please specify the argument type

File: src/app/service/eco-news/create-eco-news.service.ts [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CreateEcoNewsService {
+
+  constructor(private http: HttpClient) { }
+
+  public translations: object = {
```
Comment (nabermi2020) [inline src/app/service/eco-news/create-eco-news.service.ts]: Specify type

File: src/app/service/eco-news/eco-news.service.ts [hunk 21]
```diff
@@ -56,12 +56,12 @@ export class EcoNewsService {
       .splice(0, 3);
     this.sortedLastThreeNews.next(separetedNews);
   }
-
+
```
Comment (nabermi2020) [inline src/app/service/eco-news/eco-news.service.ts]: Please revert these changes, 'cause this functionality is out of the scope of your ticket

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 22]
```diff
@@ -1,18 +1,111 @@
...
+    }
+  }
+  
+  private navigateByUrl(): void {
+    this.router.navigateByUrl('/news');
+  }
+
+  public onSourceChange(): void {
+    this.createNewsForm.get('source').valueChanges.subscribe(source => {
+      if( !source ) {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: bad formatting

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 23]
```diff
@@ -1,18 +1,111 @@
...
+    );
+  }
+
+  public addFilters(filter: any): void { 
+    if(!filter.isActive) {
+      filter.isActive = true;
+      this.createNewsForm.value.tags.push(filter.name.toLowerCase());
+    } else {
+      filter.isActive = false;
+      this.createNewsForm.value.tags.forEach((item, index) => {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: Please extract this functionality in the separate method

File: src/app/component/general/eco-news/create-news/create-news.component.ts [hunk 24]
```diff
@@ -1,18 +1,111 @@
...
+        if (item.toLowerCase() === filter.name.toLowerCase()) {
+          this.createNewsForm.value.tags.splice(index, 1);
+        }
+      })
+    }
+  }
+
+  public setFormValue(): void {
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/create-news/create-news.component.ts]: setter should have some argument, but you don't have it, so rename it like bind********

File: src/app/service/eco-news/create-eco-news.service.ts [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    'en': {
+      text: "",
+      title: ""
+    }
+  };
+
+  private url: string = 'https://greencity.azurewebsites.net/econews';
+  private accessToken: string = localStorage.getItem('accessToken');
+  private httpOptions: any = {
```
Comment (nabermi2020) [inline src/app/service/eco-news/create-eco-news.service.ts]: You can delete type declaration here, 'cause you create new HttpHeaders instance

File: src/app/service/eco-news/create-eco-news.service.ts [hunk 26]
```diff
@@ -0,0 +1,67 @@
...
+    return this.translations[lang];
+  }
+
+  public setTranslationByLang(language: string, translations: CreateNewsInterface): void {
+    this.translations[language].text = translations.text;
+    this.translations[language].title = translations.title;
+  }
+
+  public sendFormData(form, language): Observable<any> {
+    const body: any = {
```
Comment (nabermi2020) [inline src/app/service/eco-news/create-eco-news.service.ts]: Please create a separate interface for 'body' and specify it

File: src/app/component/general/eco-news/create-news/create-news-interface.ts (hunk 11 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news-interface.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.css (hunk 12 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.css]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 13 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 14 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 15 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 16 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 18 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 17 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 19 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/service/eco-news/create-eco-news.service.ts (hunk 20 above)
Comment (Gusvarov) [reply src/app/service/eco-news/create-eco-news.service.ts]: fixed

File: src/app/service/eco-news/eco-news.service.ts (hunk 21 above)
Comment (Gusvarov) [reply src/app/service/eco-news/eco-news.service.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 22 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 24 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed

File: src/app/service/eco-news/create-eco-news.service.ts (hunk 25 above)
Comment (Gusvarov) [reply src/app/service/eco-news/create-eco-news.service.ts]: fixed

File: src/app/service/eco-news/create-eco-news.service.ts (hunk 26 above)
Comment (Gusvarov) [reply src/app/service/eco-news/create-eco-news.service.ts]: fixed

File: src/app/component/general/eco-news/create-news/create-news.component.ts (hunk 23 above)
Comment (Gusvarov) [reply src/app/component/general/eco-news/create-news/create-news.component.ts]: fixed
